authorize_test.go

```diff
@@ -71,435 +71,435 @@ func TestIsAuthorized(t *testing.T) {
-			Resource:  NewEntityUID("table", "whatever"),
-			Context:   Record{},
+			Resource:  cedar.NewEntityUID("table", "whatever"),
+			Context:   cedar.Record{},
 			Want:      true,
 			DiagErr:   1,
 		},
 		{
 			Name:      "permit-requires-context-success",
 			Policy:    `permit(principal,action,resource) when { context.x == 42 };`,
-			Entities:  Entities{},
+			Entities:  cedar.Entities{},
 			Principal: cuzco,
 			Action:    dropTable,
-			Resource:  NewEntityUID("table", "whatever"),
-			Context:   NewRecord(RecordMap{"x": Long(42)}),
+			Resource:  cedar.NewEntityUID("table", "whatever"),
+			Context:   cedar.NewRecord(cedar.RecordMap{"x": cedar.Long(42)}),
 			Want:      true,
 			DiagErr:   0,
 		},
 		{
 			Name:      "permit-requires-context-fail",
 			Policy:    `permit(principal,action,resource) when { context.x == 42 };`,
-			Entities:  Entities{},
+			Entities:  cedar.Entities{},
 			Principal: cuzco,
 			Action:    dropTable,
-			Resource:  NewEntityUID("table", "whatever"),
-			Context:   NewRecord(RecordMap{"x": Long(43)}),
+			Resource:  cedar.NewEntityUID("table", "whatever"),
+			Context:   cedar.NewRecord(cedar.RecordMap{"x": cedar.Long(43)}),
 			Want:      false,
 			DiagErr:   0,
 		},
 		{
 			Name:   "permit-requires-entities-success",
 			Policy: `permit(principal,action,resource) when { principal.x == 42 };`,
-			Entities: Entities{
-				cuzco: &Entity{
+			Entities: cedar.Entities{
+				cuzco: &cedar.Entity{
 					UID:        cuzco,
-					Attributes: NewRecord(RecordMap{"x": Long(42)}),
+					Attributes: cedar.NewRecord(cedar.RecordMap{"x": cedar.Long(42)}),
 				},
 			},
 			Principal: cuzco,
 			Action:    dropTable,
-			Resource:  NewEntityUID("table", "whatever"),
-			Context:   Record{},
+			Resource:  cedar.NewEntityUID("table", "whatever"),
+			Context:   cedar.Record{},
 			Want:      true,
 			DiagErr:   0,
 		},
 		{
 			Name:   "permit-requires-entities-fail",
 			Policy: `permit(principal,action,resource) when { principal.x == 42 };`,
-			Entities: Entities{
-				cuzco: &Entity{
+			Entities: cedar.Entities{
+				cuzco: &cedar.Entity{
 					UID:        cuzco,
-					Attributes: NewRecord(types.RecordMap{"x": Long(43)}),
+					Attributes: cedar.NewRecord(cedar.RecordMap{"x": cedar.Long(43)}),
 				},
 			},
 			Principal: cuzco,
 			Action:    dropTable,
-			Resource:  NewEntityUID("table", "whatever"),
-			Context:   Record{},
+			Resource:  cedar.NewEntityUID("table", "whatever"),
+			Context:   cedar.Record{},
 			Want:      false,
 			DiagErr:   0,
 		},
 		{
 			Name:   "permit-requires-entities-parent-success",
 			Policy: `permit(principal,action,resource) when { principal in parent::"bob" };`,
-			Entities: Entities{
-				cuzco: &Entity{
+			Entities: cedar.Entities{
+				cuzco: &cedar.Entity{
 					UID:     cuzco,
-					Parents: []EntityUID{types.NewEntityUID("parent", "bob")},
+					Parents: cedar.NewEntityUIDSetFromSlice([]cedar.EntityUID{cedar.NewEntityUID("parent", "bob")}),
```

Comment:
Here's an example of the breaking change.

I'll admit that it's a bit of a mouthful...

Reply:
Maybe receive a variadic, to cut out the `[]cedar.EntityUID{` ?  Though TBH, may not matter too much since it's going to be usually handled via JSON marshalling.

Maybe the words `FromSlice` aren't really needed.